tests/e2e/specs/shopper/mini-cart.test.js

```diff
@@ -94,4 +123,89 @@ describe( 'Shopper → Mini Cart', () => {
 			);
 		} );
 	} );
+
+	describe( 'Filled mini cart', () => {
+		beforeAll( async () => {
+			await shopper.emptyCart();
+		} );
+
+		afterEach( async () => {
+			await shopper.emptyCart();
+		} );
+
+		it( 'The Mini Cart title shows correct amount', async () => {
+			await page.click(
+				'.wc-block-grid__product:first-child .add_to_cart_button'
+			);
+
+			await expect( page ).toMatchElement( '.wc-block-mini-cart__title', {
+				text: 'Your cart (1 item)',
+			} );
+
+			await page.mouse.click( 50, 200 );
+
+			await page.click(
+				'.wc-block-grid__product:last-child .add_to_cart_button'
+			);
+
+			await expect( page ).toMatchElement( '.wc-block-mini-cart__title', {
+				text: 'Your cart (2 items)',
+			} );
+		} );
+
+		it( 'The Mini Cart products table show added products', async () => {
+			const products = await page.$$(
+				'.wc-block-all-products .wc-block-grid__product'
+			);
+			if ( products.length === 0 ) {
+				return;
+			}
```

Comment:
wouldn't this mask potential issue in test setup where there would be no products even though there should be?

Reply:
Thrown an error instead. Thanks for the head-up!